vms/platformvm/state/state.go

```diff
@@ -144,6 +158,12 @@ type State interface {
 	// Discard uncommitted changes to the database.
 	Abort()
 
+	// Asynchronously removes rejected blocks from disk and indexes accepted blocks
```

Comment:
Being super pedantic - this function isn't really async... (it doesn't spin up any new go-routines)... It supports being called async from the engine thread...

idk If there is a clearer way to say this... I'm also not super against the current wording - it just seemed a bit off to me